PlastomeRegionBurstAndAlign.py

```diff
@@ -340,165 +322,154 @@ def main(args):
-    log.info("%s" % action)
-    ###
+
+def remove_user_defined_genes(main_d_nucl, main_d_prot, exclude_list, select_mode, log):
     if exclude_list:
         if select_mode == 'int':
             to_be_excluded = [i + "_intron1" for i in exclude_list] + [i + "_intron2" for i in exclude_list]
             exclude_list = to_be_excluded
         for excluded in exclude_list:
             if excluded in main_d_nucl:
                 del main_d_nucl[excluded]
-                if select_mode == 'cds':
+                if select_mode == 'cds' and main_d_prot:
                     del main_d_prot[excluded]
             else:
                 log.warning("Region `%s` to be excluded but unable to be found in infile." % excluded)
                 pass
-    # --------------------------------------------------------------------------#
-    # EXTRACTING AND COMBINING NUCLEOTIDE SEQUENCES GENEWISE
+
+def extract_and_combine_nucleotide_sequences(main_d_nucl, out_dir):
     action = "extracting and combining nucleotide sequences genewise"
     log.info("%s" % action)
-    ###
+
+    for k, v in main_d_nucl.items():
+        out_fn_unalign_nucl = os.path.join(out_dir, 'nucl_' + k + '.unalign.fasta')
+        with open(out_fn_unalign_nucl, 'w') as hndl:
+            SeqIO.write(v, hndl, 'fasta')
+
+def multiple_sequence_alignment_nucleotide(main_d_nucl, out_dir):
+    action = "conducting multiple sequence alignment based on nucleotide sequence data"
+    log.info("%s" % action)
+
     if main_d_nucl.items():
         for k, v in main_d_nucl.items():
             out_fn_unalign_nucl = os.path.join(out_dir, 'nucl_' + k + '.unalign.fasta')
-            with open(out_fn_unalign_nucl, 'w') as hndl:
-                Bio.SeqIO.write(v, hndl, 'fasta')
+            out_fn_aligned_nucl = os.path.join(out_dir, 'nucl_' + k + '.aligned.fasta')
+            num_threads = 1
+            alignm_cmdlexec = Applications.MafftCommandline(input=out_fn_unalign_nucl, adjustdirection=True, thread=num_threads)
+            stdout, stderr = alignm_cmdlexec()
+            with open(out_fn_aligned_nucl, 'w') as hndl:
+                hndl.write(stdout)
+    else:
+        log.critical("No items in nucleotide main dictionary to process")
+        raise Exception()
 
-    # --------------------------------------------------------------------------#
-    # MULTIPLE SEQUENCE ALIGNMENT BASED ON NUCLEOTIDE SEQUENCE DATA
-    if not select_mode == 'cds':
-        action = "conducting multiple sequence alignment based on nucleotide sequence data"
-        log.info("%s" % action)
-        ###
-        if main_d_nucl.items():
-            for k, v in main_d_nucl.items():
-                out_fn_unalign_nucl = os.path.join(out_dir, 'nucl_' + k + '.unalign.fasta')
-                out_fn_aligned_nucl = os.path.join(out_dir, 'nucl_' + k + '.aligned.fasta')
-                # import subprocess
-                # subprocess.call(['mafft', '--auto', out_fn_unalign_nucl, '>', out_fn_aligned_nucl])
-
```

Comment:
Please keep these comments in the code for now. They are important for further development.

                ## TO DO ##
                # Automatically determine number of threads available #
                # Have the number of threads saved as num_threads